virtcontainers/utils/utils.go

```diff
@@ -200,3 +221,74 @@ func BuildSocketPath(elements ...string) (string, error) {
 
 	return result, nil
 }
+
+// SupportsVsocks returns true if vsocks are supported, otherwise false
+func SupportsVsocks() bool {
+	if _, err := os.Stat(vSockDevicePath); err != nil {
+		return false
+	}
+
+	if _, err := os.Stat(vHostVSockDevicePath); err != nil {
+		return false
+	}
+
+	return true
+}
+
+// FindContextID finds a unique context ID by generating a random number between 3 and max unsigned int (maxUint).
+// Using the ioctl VHOST_VSOCK_SET_GUEST_CID, findContextID asks to the kernel if the given
+// context ID (N) is available, when the context ID is not available, incrementing by 1 findContextID
+// iterates from N to maxUint until an available context ID is found, otherwise decrementing by 1
+// findContextID iterates from N to 3 until an available context ID is found, this is the last chance
+// to find a context ID available.
+// On success vhost file and a context ID greater or equal than 3 are returned, otherwise 0 and an error are returned.
+// vhost file can be used to send vhost file decriptor to QEMU.
+//
+// Benefits of using random context IDs:
+// - Reduce the probability of a *DoS attack*, since other processes don't know whatis the initial context ID
+//   used by findContextID to find a context ID available
+//
+func FindContextID() (*os.File, uint32, error) {
```

Comment:
I find this function rather difficult to read - could you maybe space the code out a bit and add some more comments?